src/psyclone/psyGen.py

```diff
@@ -3708,9 +3704,59 @@ def __init__(self, call, arg_info, access):
             # There are unit-tests where we create Arguments without an
             # associated call.
             if self._call:
+
+                # Keep original list of arguments
+                previous_arguments = \
+                        self._call.root.symbol_table.argument_list
+
+                # Find the tag to use
                 tag = "AlgArgs_" + self._text
-                self._name = self._call.root.symbol_table.symbol_from_tag(
-                    tag, self._orig_name).name
+
+                # Prepare the Argument Interface Access value
+                if access and access.name == "READ":
+                    argument_access = ArgumentInterface.Access.READ
+                elif access and access.name == "WRITE":
+                    argument_access = ArgumentInterface.Access.WRITE
+                else:
+                    # If access is READWRITE, INC, SUM, UNKNOWN or no access
+                    # is given, use a READWRITE argument interface.
+                    argument_access = ArgumentInterface.Access.READWRITE
+
+                # Find the tag or create a new symbol with expected attributes
+                new_argument = self._call.root.symbol_table.symbol_from_tag(
```

Comment:
Presumably `root` gives the Schedule of the Invoke?

Reply:
yes, I made it more explicit now with `symtab = self._call.ancestor(InvokeSchedule).symbol_table` just in case we add a top module at some point.